File: npm-packages/cordova-plugin-meteor-webapp/src/ios/AssetManifest.swift
struct AssetManifest {
  struct Entry {
    let filePath: String
    let URLPath: String
    let fileType: String
    let cacheable: Bool
    let hash: String?
    let sourceMapPath: String?
    let sourceMapURLPath: String?
  }

  let version: String
  let cordovaCompatibilityVersion: String
  var entries: [Entry]

  init(fileURL: URL) throws {
    try self.init(data: try Data(contentsOf: fileURL, options: []))
  }

  init(data: Data) throws {
    let JSON: JSONObject
    do {
      JSON = try JSONSerialization.jsonObject(with: data, options: []) as! JSONObject
    } catch {
      throw WebAppError.invalidAssetManifest(reason: "Error parsing asset manifest", underlyingError: error)
    }

    if let format = JSON["format"] as? String, format != "web-program-pre1" {
      throw WebAppError.invalidAssetManifest(reason: "The asset manifest format is incompatible: \(format)", underlyingError: nil)
    }

    guard let version = JSON["version"] as? String else {
      throw WebAppError.invalidAssetManifest(reason: "Asset manifest does not have a version", underlyingError: nil)
    }
    
    self.version = version
    
    guard let cordovaCompatibilityVersions = JSON["cordovaCompatibilityVersions"] as? JSONObject,
      let cordovaCompatibilityVersion = cordovaCompatibilityVersions["ios"] as? String else {
      throw WebAppError.invalidAssetManifest(reason: "Asset manifest does not have a cordovaCompatibilityVersion", underlyingError: nil)
    }
    
    self.cordovaCompatibilityVersion = cordovaCompatibilityVersion
    
    let entriesJSON = JSON["manifest"] as? [JSONObject] ?? []
    entries = []
    for entryJSON in entriesJSON {
      if entryJSON["where"] as? String != "client" { continue }

      if let URLPath = entryJSON["url"] as? String,
        let filePath = entryJSON["path"] as? String,
        let fileType = entryJSON["type"] as? String,
        let hash = entryJSON["hash"] as? String,
        let cacheable = entryJSON["cacheable"] as? Bool {
          let sourceMapPath = entryJSON["sourceMap"] as? String
          let sourceMapURLPath = entryJSON["sourceMapUrl"] as? String

          let entry = Entry(filePath: filePath, URLPath: URLPath,
            fileType: fileType, cacheable: cacheable, hash: hash,
            sourceMapPath: sourceMapPath, sourceMapURLPath: sourceMapURLPath)
          entries.append(entry)
      }
    }
  }
}

File: npm-packages/meteor-capacitor/android/src/main/java/com/meteorjs/capacitor/AssetManifest.java
/*
 * Vendored from https://github.com/banjerluke/capacitor-meteor-webapp
 * Copyright 2025 Luke Abbott. MIT license. See LICENSES/banjerluke-capacitor-meteor-webapp.txt.
 */

package com.meteorjs.capacitor;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

final class AssetManifest {

    static final class Entry {
        final String filePath;
        final String urlPath;
        final String fileType;
        final boolean cacheable;
        final String hash;
        final String sourceMapFilePath;
        final String sourceMapUrlPath;

        Entry(
            String filePath,
            String urlPath,
            String fileType,
            boolean cacheable,
            String hash,
            String sourceMapFilePath,
            String sourceMapUrlPath
        ) {
            this.filePath = filePath;
            this.urlPath = urlPath;
            this.fileType = fileType;
            this.cacheable = cacheable;
            this.hash = hash;
            this.sourceMapFilePath = sourceMapFilePath;
            this.sourceMapUrlPath = sourceMapUrlPath;
        }
    }

    final String version;
    final String cordovaCompatibilityVersion;
    final List<Entry> entries;

    AssetManifest(String jsonString) throws WebAppError {
        try {
            JSONObject json = new JSONObject(jsonString);
            String format = json.optString("format", null);
            if (format != null && !"web-program-pre1".equals(format)) {
                throw new WebAppError(
                    WebAppError.Type.INVALID_ASSET_MANIFEST,
                    "The asset manifest format is incompatible: " + format
                );
            }

            version = json.optString("version", null);
            if (version == null) {
                throw new WebAppError(
                    WebAppError.Type.INVALID_ASSET_MANIFEST,
                    "Asset manifest does not have a version"
                );
            }

            JSONObject compatibility = json.optJSONObject("cordovaCompatibilityVersions");
            if (compatibility == null) {
                throw new WebAppError(
                    WebAppError.Type.INVALID_ASSET_MANIFEST,
                    "Asset manifest does not have cordovaCompatibilityVersions"
                );
            }

            cordovaCompatibilityVersion = compatibility.optString("android", null);
            if (cordovaCompatibilityVersion == null) {
                throw new WebAppError(
                    WebAppError.Type.INVALID_ASSET_MANIFEST,
                    "Asset manifest does not have a cordovaCompatibilityVersion for android"
                );
            }

            JSONArray manifestEntries = json.optJSONArray("manifest");
            if (manifestEntries == null) {
                entries = Collections.emptyList();
                return;
            }

            List<Entry> parsedEntries = new ArrayList<>(manifestEntries.length());
            for (int i = 0; i < manifestEntries.length(); i++) {
                JSONObject entryJson = manifestEntries.getJSONObject(i);
                if (!"client".equals(entryJson.optString("where", null))) {
                    continue;
                }

                String filePath = entryJson.getString("path");
                String urlPath = entryJson.getString("url");
                String fileType = entryJson.getString("type");
                boolean cacheable = entryJson.getBoolean("cacheable");
                String hash = entryJson.optString("hash", null);
                String sourceMapFilePath = entryJson.optString("sourceMap", null);
                String sourceMapUrlPath = entryJson.optString("sourceMapUrl", null);

                parsedEntries.add(
                    new Entry(
                        filePath,
                        urlPath,
                        fileType,
                        cacheable,
                        hash,
                        sourceMapFilePath,
                        sourceMapUrlPath
                    )
                );
            }
            entries = Collections.unmodifiableList(parsedEntries);
        } catch (JSONException e) {
            throw new WebAppError(
                WebAppError.Type.INVALID_ASSET_MANIFEST,
                "Error parsing asset manifest",
                e
            );
        }
    }
}
